### PR TITLE
PRMDR_687 - Updated createDocumentReference to push to staging bucket

### PR DESCRIPTION
**Pull request by SRAlexander**

### CODE REVIEW

File: lambdas/services/create_document_reference_service.py [hunk 1]
```diff
@@ -11,108 +11,136 @@
...
+                (document_reference, doc_type) = self.prepare_doc_object(
+                    nhs_number, document
+                )
+                self.check_valid_doc_type(doc_type)
+
+                if doc_type == SupportedDocumentTypes.ARF.value:
+                    arf_documents.append(document_reference)
+                    arf_documents_dict_format.append(document_reference.to_dict())
+
+                if doc_type == SupportedDocumentTypes.LG.value:
```
Comment (NogaNHS) [inline lambdas/services/create_document_reference_service.py:58]: if else here? no need to check for both as you can only type for a doc

File: lambdas/services/create_document_reference_service.py [hunk 2]
```diff
@@ -11,108 +11,136 @@
...
+            validate_nhs_number(nhs_number)
             for document in documents_list:
-                document_reference = self.prepare_doc_object(document)
-                self.prepare_pre_signed_url(document_reference)
-            if self.lg_documents:
-                validate_lg_files(self.lg_documents, self.nhs_number)
+                (document_reference, doc_type) = self.prepare_doc_object(
+                    nhs_number, document
+                )
+                self.check_valid_doc_type(doc_type)
```
Comment (NogaNHS) [inline lambdas/services/create_document_reference_service.py:52]: Is this method necessary as we're checking the types again on lines 54-58?

File: lambdas/services/create_document_reference_service.py [hunk 3]
```diff
@@ -11,108 +11,136 @@
...
             raise CreateDocumentRefException(400, LambdaError.CreateDocNoType)
 
-        logger.info("Provided document is supported")
-
-        (
-            s3_destination,
-            documents_type_list,
-            documents_list_dict_format,
-        ) = self.return_info_by_doc_type(document_type)
+        logger.info(PROVIDED_DOCUMENT_SUPPORTED_MESSAGE)
```
Comment (NogaNHS) [inline lambdas/services/create_document_reference_service.py:112]: Should this be moved as we're not checking the type here anymore?

File: lambdas/services/create_document_reference_service.py [hunk 4]
```diff
@@ -11,108 +11,136 @@
...
+                if doc_type == SupportedDocumentTypes.LG.value:
+                    lg_documents.append(document_reference)
+                    lg_documents_dict_format.append(document_reference.to_dict())
+
+                url_responses[
+                    document_reference.file_name
+                ] = self.prepare_pre_signed_url(document_reference)
+
+            if lg_documents:
+                validate_lg_files(lg_documents)
```
Comment (NogaNHS) [inline lambdas/services/create_document_reference_service.py:67]: This needs the nhs number passed into it.

**Review by NogaNHS — COMMENTED** [review]
Need rebasing with main

File: lambdas/services/create_document_reference_service.py (hunk 2 above)
Comment (SRAlexander) [reply lambdas/services/create_document_reference_service.py:52]: I'm going with a yes, it's a fail fast condition and reusable logic.

File: lambdas/services/create_document_reference_service.py (hunk 4 above)
Comment (SRAlexander) [reply lambdas/services/create_document_reference_service.py]: Added

File: lambdas/services/create_document_reference_service.py (hunk 2 above)
Comment (SRAlexander) [reply lambdas/services/create_document_reference_service.py]: I was going to say yes but I've now extracted it into the default section of a switch statement

File: lambdas/services/create_document_reference_service.py (hunk 1 above)
Comment (SRAlexander) [reply lambdas/services/create_document_reference_service.py]: Used a switch instead, tidier than else-if

**Review by NogaNHS — APPROVED** [review]

**Pull request merged**